### ISSUE TITLE
verbose logging of "Run took %fms" is causing performance impact on the default configuration and a bad user experience on docker

### ISSUE THREAD

**Issue by filipecosta90** [RedisAI/RedisAI#240]
If running RedisAI for the first time, or with a verbose log level ( default on RedisAI image (( non containerized redis default is notice )))
```
docker run -p 6379:6379 redisai/redisai:latest
```
we receive an enormous amount of logs, as seen on the example bellow. Considering that the information of backend inference duration will be very important in the future, we should not remove the calculation but adjust the log level in accordance or ( even better ) keep the information but dont log it ( and provide it in the future on the info command or similar )

```
1:C 05 Nov 2019 04:16:28.947 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 05 Nov 2019 04:16:28.947 # Redis version=5.0.5, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 05 Nov 2019 04:16:28.947 # Configuration loaded
1:M 05 Nov 2019 04:16:28.948 * Running mode=standalone, port=6379.
1:M 05 Nov 2019 04:16:28.948 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
1:M 05 Nov 2019 04:16:28.948 # Server initialized
1:M 05 Nov 2019 04:16:28.948 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
1:M 05 Nov 2019 04:16:29.077 * Module 'ai' loaded from /usr/lib/redis/modules/redisai.so
1:M 05 Nov 2019 04:16:29.077 * Ready to accept connections
2019-11-05 04:16:53.596371: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
1:M 05 Nov 2019 04:17:32.446 * <ai> RAI_ModelRun took 0.050000s
1:M 05 Nov 2019 04:17:32.447 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.448 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.450 * <ai> RAI_ModelRun took 0.001000s
1:M 05 Nov 2019 04:17:32.451 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.452 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.454 * <ai> RAI_ModelRun took 0.001000s
1:M 05 Nov 2019 04:17:32.455 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.456 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.458 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.459 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.460 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.461 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.463 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.464 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.465 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.467 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.468 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.470 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.471 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.472 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.474 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.475 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.477 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.478 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.480 * <ai> RAI_ModelRun took 0.001000s
1:M 05 Nov 2019 04:17:32.481 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.482 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.484 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.486 * <ai> RAI_ModelRun took 0.001000s
1:M 05 Nov 2019 04:17:32.487 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.489 * <ai> RAI_ModelRun took 0.001000s
1:M 05 Nov 2019 04:17:32.490 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.491 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.493 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.494 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.496 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.497 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.499 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.501 * <ai> RAI_ModelRun took 0.001000s
1:M 05 Nov 2019 04:17:32.502 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.504 * <ai> RAI_ModelRun took 0.001000s
1:M 05 Nov 2019 04:17:32.505 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.507 * <ai> RAI_ModelRun took 0.001000s
1:M 05 Nov 2019 04:17:32.508 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.509 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.511 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.512 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.514 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.515 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.517 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.518 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.520 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.521 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.523 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.524 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.526 * <ai> RAI_ModelRun took 0.001000s
1:M 05 Nov 2019 04:17:32.527 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.528 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.530 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.531 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.533 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.534 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.536 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.537 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.539 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.540 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.542 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.544 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.545 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.546 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.548 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.549 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.551 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.552 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.554 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.556 * <ai> RAI_ModelRun took 0.001000s
1:M 05 Nov 2019 04:17:32.557 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.559 * <ai> RAI_ModelRun took 0.001000s
1:M 05 Nov 2019 04:17:32.560 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.562 * <ai> RAI_ModelRun took 0.001000s
1:M 05 Nov 2019 04:17:32.563 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.565 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.566 * <ai> RAI_ModelRun took 0.000000s
1:M 05 Nov 2019 04:17:32.568 * <ai> RAI_ModelRun took 0.001000s
```

**Comment by lantiga** [comment]
@filipecosta90 Thank you. I was actually looking into the AI info API for modules, it looks like it will fit our bill. This would allow us to just use the `INFO` command to log execution times, avoiding `AI.INFO`, which has some advantages for integrations that already use the `INFO` command.

**Comment by filipecosta90** [comment]
looking at the model info PR on redis that is already merged ( https://github.com/antirez/redis/pull/6270 ) we will still need to keep that info within some context of RedisAI which will then be used to reported back in a new method with the following signature:
`
void InfoFunc(RedisModuleInfoCtx *ctx, int for_crash_report)
`
with that in mind I would suggest adding ( at least ) two fields two the `RAI_Model` struct in the same manner that redis does to have stats:
```
long long backend_calls;
long long backend_microseconds;
```
afterwards, when someone called `INFO` we would report for each model the average backend_duration. basically `RAI_Model` would look like:
```
typedef struct RAI_Model {
  void* model;
  // TODO: use session pool? The ideal would be to use one session per client.
  //       If a client disconnects, we dispose the session or reuse it for
  //       another client.
  void *session;
  RAI_Backend backend;
  char* devicestr;
  char **inputs;
  size_t ninputs;
  char **outputs;
  size_t noutputs;
  long long refCount;
  long long backend_calls;
  long long backend_microseconds;
  void* data;
} RAI_Model;
```
what do you think @lantiga ? This would be the minimal to expected metric to have. The holly gral would be to have something that would report not singleton metrics ( like a latency histogram ), but I would go for the simple solution now and remove the LogLine in exchange to keeping the info on the Model and reporting the average when requested.

**Comment by filipecosta90** [comment]
solved by #248
